### PR TITLE
refactor(code-refactor): :sparkles: Refactored code to resolve #292 issue

### LINKED ISSUE #292: Make common FCC colors constants
In our project we are using the colors defined in https://design-style-guide.freecodecamp.org/ such as #0a0a23

We are repeating the process of typing out these hex codes often in code, ideally we should define the colors once as in https://tailwindcss.com/docs/customizing-colors#using-custom-colors as string like "gray-90" and not be typing out hex codes manually.

Completion of this issue requires:

1.) Define custom colors
2.) Refactor all code to use the defined colors (can be done in multiple PRs if needed)

### PR DESCRIPTION
**Pull request by HyperDanisH**
I have added colors variables to the tailwind.config.js file, now we can access the values without having to hardcode them each time. We can still have all the screen sizes, font sizes etc there but we will begin by colors only. I thought this could be a right fix. I will appericiate any review on this.

Todos:
- [x] Remove all hardcode color values and fill those with variables defined in tailwind.config.js file.

#292

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

### CODE REVIEW

**Comment by GuillermoFloresV** [comment]
Hi, @HyperDanisH thanks for taking on this issue. Can you explain the gray naming convention? We can move forward regarding code review once we have that established.

File: tailwind.config.js [hunk 1]
```diff
@@ -3,7 +3,36 @@ module.exports = {
     './pages/**/*.{js,ts,jsx,tsx}',
     './components/**/*.{js,ts,jsx,tsx}'
   ],
-  theme: {},
+  theme: {
+    colors: {
+      transparent: 'transparent',
+      current: 'currentColor',
+      gray: {
+        0o0: '#ffffff',
```
Comment (roberthuskins) [inline tailwind.config.js:11]: Why 0o0 instead of 00?

File: tailwind.config.js [hunk 2]
```diff
@@ -3,7 +3,36 @@ module.exports = {
     './pages/**/*.{js,ts,jsx,tsx}',
     './components/**/*.{js,ts,jsx,tsx}'
   ],
-  theme: {},
+  theme: {
```
Comment (roberthuskins) [inline tailwind.config.js:6]: Add comment with link to style guide: https://design-style-guide.freecodecamp.org/

File: tailwind.config.js [hunk 3]
```diff
@@ -3,7 +3,36 @@ module.exports = {
...
+        10: '#dfdfe2',
+        15: '#d0d0d5',
+
+        // DARK
+        75: '#3b3b4f',
+        80: '#2a2a40',
+        85: '#1b1b32',
+        90: '#0a0a23'
+      },
+      primary: {
```
Comment (roberthuskins) [inline tailwind.config.js:22]: My preference would be to prefix with FCC. What do you think @GuillermoFloresV @utsab 

ex:
fcc-gray-90
fcc-primary-purple

File: tailwind.config.js (hunk 1 above)
Comment (HyperDanisH) [reply tailwind.config.js:11]: Hey, thanks for getting back, it is due to error caused due to octal numbers not allowed in strict mode.

File: tailwind.config.js (hunk 3 above)
Comment (HyperDanisH) [reply tailwind.config.js:22]: We can discuss and proceed it that way.

Comment (GuillermoFloresV) [reply tailwind.config.js]: Yeah, I think this works well. That way new contributors know that the color's name came directly from FCC.

**Comment by GuillermoFloresV** [comment]
> > Hi, @HyperDanisH thanks for taking on this issue. Can you explain the gray naming convention? We can move forward regarding code review once we have that established.
> 
> This was accessed in tailwind by bg-gray-0, bg-gray-80, etc. Although as @roberthuskins has recommended, I have made it bg-fcc-gray-90 rather than bg-gray-90. Similarly, primary colors as "fcc-primary-purple" and secondary as "fcc-secondary-darkPurple".

Thanks, this makes it less ambiguous for new contributors using these colors.

**Comment by HyperDanisH** [comment]
> > > Hi, @HyperDanisH thanks for taking on this issue. Can you explain the gray naming convention? We can move forward regarding code review once we have that established.
> > 
> > 
> > This was accessed in tailwind by bg-gray-0, bg-gray-80, etc. Although as @roberthuskins has recommended, I have made it bg-fcc-gray-90 rather than bg-gray-90. Similarly, primary colors as "fcc-primary-purple" and secondary as "fcc-secondary-darkPurple".
> 
> Thanks, this makes it less ambiguous for new contributors using these colors.

Hi,
When are we going to proceed with the merging of this PR? Is more work required? And if so then how can I do it.

**Comment by GuillermoFloresV** [comment]
> > > > Hi, @HyperDanisH thanks for taking on this issue. Can you explain the gray naming convention? We can move forward regarding code review once we have that established.
> > > 
> > > 
> > > This was accessed in tailwind by bg-gray-0, bg-gray-80, etc. Although as @roberthuskins has recommended, I have made it bg-fcc-gray-90 rather than bg-gray-90. Similarly, primary colors as "fcc-primary-purple" and secondary as "fcc-secondary-darkPurple".
> > 
> > 
> > Thanks, this makes it less ambiguous for new contributors using these colors.
> 
> Hi, When are we going to proceed with the merging of this PR? Is more work required? And if so then how can I do it.

Hi, sorry for the delay. We will get to fully review this PR soon. If there are any other action items once our review is complete, we will make sure to let you know. Thanks for your patience.

**Comment by GuillermoFloresV** [comment]
Hi @HyperDanisH , I took a look at your fork and found that a lot of our components are no longer following our desired designs. Would you be able to investigate why this is happening? I have linked some examples into this [Google Doc](https://docs.google.com/document/d/1Xjkq1juWTtzDSCW3a1znROa2nuxpoepbkbbmkpvUxvQ/edit?usp=sharing). Please let me know if you have any questions.

**Comment by HyperDanisH** [comment]
Hi @GuillermoFloresV, Now that you have mentioned this issue, I found out there are some colors that are used differently than defined in freecodecamp style guide. There are two ways we can approach this, one is to define those colors as globals, second is replace those colors in ui to which are defined in style guide. I would like to have some suggestions in this case here.

Best,

**Comment by GuillermoFloresV** [comment]
@HyperDanisH Great! Thanks. Looking at your branch I can see that the colors are now working as intended. 

I noticed that we are getting these warnings inside of the console when the app ran.

image

I've gone ahead and suppressed those warnings in my own commit to eliminate another round of back and forth talks.

**Comment by GuillermoFloresV** [comment]
@utsab @roberthuskins 

Since I've added some code here, could one of you add your review to sanity-check me? Once that's done we can merge.

@HyperDanisH Thanks for your work on this PR!

**Comment by HyperDanisH** [comment]
Hi, @GuillermoFloresV, I was happy to work with you on this issue. If there is any more such issue then I am here for the contributions. At last thanks and would you mind sharing some resources about the wireframe and other stuff for this project.

Best, 
Danish

**Review by utsab — APPROVED** [review]

**Comment by GuillermoFloresV** [comment]
> Hi, @GuillermoFloresV, I was happy to work with you on this issue. If there is any more such issue then I am here for the contributions. At last thanks and would you mind sharing some resources about the wireframe and other stuff for this project.
> 
> Best, Danish

Hi, Danish, PR was merged, thanks for your help! 

As for wireframes, we have some very old ones that I can dig up for you if you'd like. Our styling and design have evolved over time since their creation and I am sure they won't be as useful as they once were. For now, you can take a look at some of the other issues that we have open and feel free to ask for clarity regarding any of them.

**Pull request merged**